x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -106,7 +106,7 @@ public static List<TestCaseSupplier> forUnaryCastingToDouble(
             i -> expected.applyAsDouble(i),
             min.intValue(),
             max.intValue(),
-            List.of()
+            warnings
```

Comment:
question: what are examples where we expect to have a list of warnings that we assert for all tested data types? In this PR, I only see an empty list of warnings everywhere.

Reply:
The idea is that we would use the `min` and `max` bounds settings to define ranges where the function returns a value, and ranges where it returns null with a warning. I haven't added any new warnings in this PR, because I'm just trying to get the shared code part merged quickly to avoid conflicts, but you can see some in the [log10 tests](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/Log10Tests.java)

Short term, meaning over the next few PRs, I'm working on https://github.com/elastic/elasticsearch/issues/98698, which will require a lot more warnings from functions, and at the moment that's blocked on these test generators being able to accept warnings and defined ranges.